File: src/main/java/com/example/home_store/Service/OrderService.java
package com.example.home_store.Service;

import com.example.home_store.DTO.CreateOrderRequestDto;
import com.example.home_store.DTO.OrderDto;

import java.util.List;
import java.util.UUID;

public interface OrderService {
    OrderDto createOrder(CreateOrderRequestDto dto);
    List<OrderDto> getMyOrders(UUID userId);
    List<OrderDto> getAllOrders();
    OrderDto updateStatus(UUID orderId, String status);
}

File: frontend/src/app/core/services/order.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

import { API_CONFIG } from '../config/api.config';
import { Order } from '../models/order.model';
import { CreateOrderRequest } from '../models/create-order-request.model';
import { UpdateOrderStatusRequest } from '../models/update-order-status-request.model';

@Injectable({
  providedIn: 'root'
})
export class OrderService {
  private http = inject(HttpClient);
  private ordersUrl = `${API_CONFIG.baseUrl}/orders`;
  private adminOrdersUrl = `${API_CONFIG.baseUrl}/admin/orders`;

  createOrder(data: CreateOrderRequest): Observable<Order> {
    return this.http.post<Order>(this.ordersUrl, data);
  }

  getMyOrders(userId: string): Observable<Order[]> {
    const params = new HttpParams().set('userId', userId);
    return this.http.get<Order[]>(`${this.ordersUrl}/my`, { params });
  }

  getAllOrders(): Observable<Order[]> {
    return this.http.get<Order[]>(this.adminOrdersUrl);
  }

  updateStatus(orderId: string, data: UpdateOrderStatusRequest): Observable<Order> {
    return this.http.patch<Order>(`${this.adminOrdersUrl}/${orderId}/status`, data);
  }
}
